pkg/server/server.go

```diff
@@ -539,6 +541,14 @@ func (s *Server) HTTPHandler() http.Handler {
 	handle("/api/console/knative-event-sources", authHandler(s.handleKnativeEventSourceCRDs))
 	handle("/api/console/knative-channels", authHandler(s.handleKnativeChannelCRDs))
 
+	// Dev-Console Proxy
+	handle(devConsoleProxyEndpoint, http.StripPrefix(
+		proxy.SingleJoiningSlash(s.BaseURL.Path, devConsoleProxyEndpoint),
+		authHandlerWithUser(func(user *auth.User, w http.ResponseWriter, r *http.Request) {
+			devconsoleProxy.Handler(w, r)
+		})),
+	)
+
```

Comment:
Can we have here a single line that delegates everything that starts with `/api/dev-console/` into our `pkg/devconsole` package and then split it there into different handlers. This makes it easier to add more methods in the future. (Similar to the knativeProxyEndpoint),

Otherwise the server.go grows and grows ... as it does already.

```go
  handle(devConsoleEndpoint, authHandlerWithUser(devConsoleHandler.Handle))
```

Reply:
Yes, the current code exactly does that. It will delegate everything that starts with `/api/dev-console/` to the devconsole package.

We can either put everything together in the function as currently done or assign each to different vars and call the vars, like the one you have suggested.